go/vt/vttablet/tabletserver/schema/engine.go

```diff
@@ -126,15 +131,62 @@ func (se *Engine) InitDBConfig(cp dbconfigs.Connector) {
 	se.cp = cp
 }
 
+// syncVTDatabase is called either the first time a primary starts or on consequent loads to possibly upgrade to a
+// new Vitess version. This is the only entry point into the sidecardb module to get the _vt database to the desired
+// schema for the running Vitess version.
+// There is some extra logging in here which can be removed in a future version (>v16) once the new schema init
+// functionality is stable.
+func (se *Engine) syncVTDatabase(ctx context.Context, conn *dbconnpool.DBConnection) error {
+	log.Infof("In syncVTDatabase")
+	defer func(start time.Time) {
+		log.Infof("syncVTDatabase took %d ms", time.Since(start).Milliseconds())
+	}(time.Now())
+
+	var exec sidecardb.Exec = func(ctx context.Context, query string, maxRows int, wantFields bool, useVT bool) (*sqltypes.Result, error) {
+		if useVT {
+			_, err := conn.ExecuteFetch(sidecardb.UseVTDatabaseQuery, maxRows, wantFields)
+			if err != nil {
+				return nil, err
+			}
+		}
+		return conn.ExecuteFetch(query, maxRows, wantFields)
+	}
+	log.Infof("before sidecardb.Init")
+	if err := sidecardb.Init(ctx, exec); err != nil {
+		log.Errorf("Error in sidecardb.Init: %+v", err)
+		// temporary logging, to be deleted in v17
+		if strings.Contains(err.Error(), "--read-only") {
```

Comment:
We should instead cast to a MySQLError and use the error codes: https://dev.mysql.com/doc/mysql-errors/en/server-error-reference.html
```
Error number: 1836; Symbol: [ER_READ_ONLY_MODE](https://dev.mysql.com/doc/mysql-errors/8.0/en/server-error-reference.html#error_er_read_only_mode); SQLSTATE: HY000

Message: Running in read-only mode
```

Looks like we'll need to add that one here: https://github.com/vitessio/vitess/blob/main/go/mysql/constants.go

Looks like this is the error you had in mind:
```
mysql> create database svt;
ERROR 1290 (HY000): The MySQL server is running with the --super-read-only option so it cannot execute this statement
```

Now that I've looked into it, we should look for any of these MySQL server error codes here: 1836, 1290, 1874:
```
Error number: 1290; Symbol: [ER_OPTION_PREVENTS_STATEMENT](https://dev.mysql.com/doc/mysql-errors/8.0/en/server-error-reference.html#error_er_option_prevents_statement); SQLSTATE: HY000

Message: The MySQL server is running with the %s option so it cannot execute this statement
```
```
Error number: 1874; Symbol: [ER_INNODB_READ_ONLY](https://dev.mysql.com/doc/mysql-errors/8.0/en/server-error-reference.html#error_er_innodb_read_only); SQLSTATE: HY000

Message: InnoDB is in read only mode.
```

We already have a const for that 1290: https://github.com/vitessio/vitess/blob/main/go/mysql/constants.go#L420

And one for 1874: https://github.com/vitessio/vitess/blob/main/go/mysql/constants.go#L428

Reply:
removed that extra debug logging, can be added in the super-read-only PR if required for debugging.